core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -416,6 +418,16 @@ private SslContextFactory createSslContextFactory() {
         sslContextFactory.setKeyManagerFactoryAlgorithm(
             config.getString(RestConfig.SSL_KEYMANAGER_ALGORITHM_CONFIG));
       }
+      if (config.getBoolean(RestConfig.SSL_KEYSTORE_RELOAD_CONFIG)) {
+        Path keystorePath = Paths.get(URI.create(sslContextFactory.getKeyStorePath()));
+        try {
+          FileWatcher.onFileChange(keystorePath, () -> 
+              sslContextFactory.reload(scf -> log.info("Reloaded SSL cert")));
+          log.info("Enabled SSL cert auto reload: " + keystorePath);
+        } catch (java.io.IOException e) {
```

Comment:
do we know when this might fail?

Reply:
As it is IOException, it means the file can not be loaded or something like that.